eclair-core/src/main/scala/fr/acinq/eclair/package.scala

```diff
@@ -159,4 +159,55 @@ package object eclair {
     * We use this in the context of timestamp filtering, when we don't need an upper bound.
     */
   val MaxEpochSeconds = Duration.fromNanos(Long.MaxValue).toSeconds
+
+  /**
+    * One MilliSatoshi is a thousand of a Satoshi, the smallest unit usable in bitcoin
+    * @param amount
+    */
+  case class MilliSatoshi(amount: Long) {
+    // @formatter:off
+    def toLong = amount
+    def +(other: MilliSatoshi) = MilliSatoshi(amount + other.amount)
+    def -(other: MilliSatoshi) = MilliSatoshi(amount - other.amount)
+    def *(m: Long) = MilliSatoshi(amount * m)
+    def /(d: Long) = MilliSatoshi(amount / d)
+    def compare(other: MilliSatoshi): Int = if (amount == other.amount) 0 else if (amount < other.amount) -1 else 1
+    def <= (that: MilliSatoshi): Boolean = compare(that) <= 0
+    def >= (that: MilliSatoshi): Boolean = compare(that) >= 0
+    def <  (that: MilliSatoshi): Boolean = compare(that) < 0
+    def >  (that: MilliSatoshi): Boolean = compare(that) > 0
+    def unary_-() = MilliSatoshi(-amount)
+    def toSatoshi: Satoshi = Satoshi(amount / 1000)
```

Comment:
conversion here is not lossless, this should be reflected in the method's name, I would suggest `truncateToSatoshi`

Reply:
Agreed and i like the name, implemented in bda3655